src/page-weekly-snapshot/WeeklyReportContext.tsx

```diff
@@ -0,0 +1,192 @@
+  fetchReferenceFacilities,
+  fetchUserFacilities,
+} from "../facilities-context/actions";
+import {
+  Facilities,
+  ReferenceFacility,
+  Scenario,
+} from "../page-multi-facility/types";
+
+export interface WeeklyReportState {
+  loading: boolean;
+  stateName: string | null;
+  scenario: Scenario | null;
+  sharedScenarios: Scenario[];
+  facilities: Facilities | ReferenceFacility[];
+}
+
+type WeeklyReportActions =
+  | SharedScenariosAction
+  | RequestActions
+  | ScenarioAction
+  | StateNameAction
+  | FacilitiesAction;
+
+type SharedScenariosAction = {
+  type: "RECEIVE_SHARED_SCENARIOS";
+  payload: Scenario[];
+};
+
+type ScenarioAction = {
+  type: "RECEIVE_SCENARIO";
+  payload: Scenario | null;
+};
+
+type FacilitiesAction = {
+  type: "RECEIVE_FACILITIES";
+  payload: Facilities | ReferenceFacility[];
+};
+
+type StateNameAction = {
+  type: "UPDATE_STATE_NAME";
+  payload: string;
+};
+
+type RequestActions = {
+  type: "REQUEST_SHARED_SCENARIOS";
+};
+
+type WeeklyReportDispatch = (action: any) => void;
+
+interface WeeklyReportContext {
+  state: WeeklyReportState;
+  dispatch: WeeklyReportDispatch;
+}
+
+const WeeklyReportContext = React.createContext<
+  WeeklyReportContext | undefined
+>(undefined);
+
+function weeklyReportReducer(
+  state: WeeklyReportState,
+  action: WeeklyReportActions,
+): WeeklyReportState {
+  switch (action.type) {
+    case "REQUEST_SHARED_SCENARIOS":
+      return Object.assign({}, state, { loading: true });
+    case "RECEIVE_SHARED_SCENARIOS":
+      return Object.assign({}, state, {
+        loading: false,
+        sharedScenarios: action.payload,
+      });
+    case "RECEIVE_SCENARIO":
+      return Object.assign({}, state, {
+        scenario: action.payload,
+      });
+    case "UPDATE_STATE_NAME":
+      return Object.assign({}, state, {
+        stateName: action.payload,
+      });
+    case "RECEIVE_FACILITIES":
+      return Object.assign({}, state, {
+        facilities: action.payload,
+      });
+    default:
+      return state;
+  }
+}
+
+export const WeeklyReportProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [state, dispatch] = React.useReducer(weeklyReportReducer, {
+    loading: false,
+    stateName: null,
+    scenario: null,
+    sharedScenarios: [],
+    facilities: [],
+  });
+
+  useEffect(() => {
+    dispatch({ type: "REQUEST_SHARED_SCENARIOS" });
+    getSharedBaselineScenarios().then((scenarios) => {
+      if (scenarios) {
+        dispatch({ type: "RECEIVE_SHARED_SCENARIOS", payload: scenarios });
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (state.stateName) {
+      getSharedScenarioByStateName(state.sharedScenarios, state.stateName).then(
+        (scenario) => {
+          dispatch({ type: "RECEIVE_SCENARIO", payload: scenario || null });
+        },
+      );
+    }
+  }, [state.stateName, state.sharedScenarios]);
+
+  useEffect(() => {
+    const systemType = "State Prison";
```

Comment:
Are we only concerned about State Prisons and not County Jails for this feature?

Reply:
yes I think so, just confirmed with @sychang !